tapiriik/services/DecathlonCoach/decathloncoach.py

```diff
@@ -0,0 +1,470 @@
+# Synchronization module for decathloncoach.com
+# (c) 2018 Charles Anssens, charles.anssens@decathlon.com
+from tapiriik.settings import WEB_ROOT, DECATHLONCOACH_CLIENT_SECRET, DECATHLONCOACH_CLIENT_ID, DECATHLONCOACH_API_KEY
+from tapiriik.services.service_base import ServiceAuthenticationType, ServiceBase
+from tapiriik.services.service_record import ServiceRecord
+from tapiriik.database import cachedb
+from tapiriik.services.interchange import UploadedActivity, ActivityType, ActivityStatistic, ActivityStatisticUnit, Waypoint, WaypointType, Location, Lap
+from tapiriik.services.api import APIException, UserException, UserExceptionType, APIExcludeActivity
+from lxml import etree
+import xml.etree.ElementTree as xml
+from django.core.urlresolvers import reverse
+from datetime import datetime, timedelta
+from urllib.parse import urlencode
+import calendar
+import requests
+import os
+import logging
+import pytz
+import re
+import time
+import json
+from dateutil.parser import parse
+
+
+logger = logging.getLogger(__name__)
+
+class DecathlonCoachService(ServiceBase):
+    ID = "decathloncoach"
+    DisplayName = "DecathlonCoach"
+    DisplayAbbreviation = "DC"
+    AuthenticationType = ServiceAuthenticationType.OAuth
+    UserProfileURL = "https://www.decathloncoach.com/fr-fr/portal/?{0}"
+    UserActivityURL = "http://www.decathloncoach.com/fr-fr/portal/activities/{1}"
+    accountOauth = "https://account.geonaute.com/oauth"
+    AuthenticationNoFrame = True  # They don't prevent the iframe, it just looks really ugly.
+    PartialSyncRequiresTrigger = False
+    LastUpload = None
+    
+    ApiEndpoint = "https://api.decathlon.net/linkdata"

```

Comment:
Thanks for https://github.com/cpfair/tapiriik/pull/417, highly appreciated and looking forward to the merge :+1: 

@canssens, I was looking for this API endpoint, but I have not found any public docs so far. Anyway, this PR demonstrates really well how to use it, so I wonder if it's possible to use this API endpoint for a sync outside of tapiriik as well until this is merged? Also, how can one create an OAuth app and get the related credentials?

Thanks, Gergely

Reply:
Hello @daroczig , yes we are opening our API.
We have not yet an API developer portal, but you can contact me at charles.anssens@decathlon.com .

Charles

Reply:
This is awesome, thanks a lot, Charles!
Sending an e-mail to you regarding the API access.